### DISCUSSION TITLE
[HxCheckBox] [HxSwitch] Tri-State feature request

### DISCUSSION THREAD

**Discussion by bluisanajmr** [havit/Havit.Blazor#648] [General]
It would be very useful if the checkbox or switch controls had a tri-state option.  I am currently needing this to be used as the header in a HxGrid where one of the columns is a boolean.  All of my grid headers allow searching but it isn't currently feasible to allow searching within a boolean column header with a switch or checkbox because that would force all records to be either True or False when initially displayed.  Ideally, the user would be able to see all records and then choose to see only true or false records by clicking the checkbox or switch.

![image](https://github.com/havit/Havit.Blazor/assets/135136836/27ce9478-5888-44f1-a6e8-3972706af511)

**Comment by hakenr** [comment]
There is no tri-state switch currently available in Bootstrap. Do you have any idea how it should look like?

I suggest you to use [`HxSelect`](https://havit.blazor.eu/components/HxSelect) with these options:
* -any-
* yes 
* no

**Comment by bluisanajmr** [comment]
Thanks Hakenr, 

My next solution was going to be a select but it would be much slicker with a checkbox or switch that behaves like this one. 

https://blazor.radzen.com/checkbox

**Comment by hakenr** [comment]
I've discovered that Bootstrap provides a built-in solution for handling indeterminate checkbox states, which can be found in their documentation here: https://getbootstrap.com/docs/5.0/forms/checks-radios/#indeterminate

While this solution addresses the issue, it's worth noting that the implementation isn't as streamlined as one might hope for. Nonetheless, it's a useful feature for certain use cases. I will put this to our backlog.
